lib/auth/auth_with_roles.go

```diff
@@ -4077,9 +4077,51 @@ func (a *ServerWithRoles) SetClusterNetworkingConfig(ctx context.Context, newNet
 		return trace.AccessDenied("proxy peering is an enterprise-only feature")
 	}
 
+	oldNetConf, err := a.authServer.GetClusterNetworkingConfig(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := a.validateCloudNetworkConfigUpdate(newNetConfig, oldNetConf); err != nil {
+		return trace.Wrap(err)
+	}
+
 	return a.authServer.SetClusterNetworkingConfig(ctx, newNetConfig)
 }
 
+func cloudTenantNetworkingError(field string) string {
+	return fmt.Sprintf("cloud tenants cannot update %q", field)
+}
+
+func (a *ServerWithRoles) validateCloudNetworkConfigUpdate(newConfig, oldConfig types.ClusterNetworkingConfig) error {
+	if a.hasBuiltinRole(types.RoleAdmin) {
+		return nil
+	}
+
+	if !modules.GetModules().Features().Cloud {
+		return nil
```

Comment:
The same as above. IMO this should be an error

Reply:
My bad, this is validate not set :facepalm: 
But in this case, if you're an admin then we don't need to validate the values? Shouldn't this and `if a.hasBuiltinRole(types.RoleAdmin) {` check be done last not first?

Reply:
As far as I understand from the issue (https://github.com/gravitational/teleport/issues/18829#issuecomment-1581448891) cluster_networking_config is going to be a dynamic resource and so we'll want admin users to be able to update it without checking the rules, I think